File: SAUCEDEMO-Playwright/Python/utils/app_factory.py
from utils.driver_factory import WebDriverManagement

class AppFactory:
    def __init__(self,wd:WebDriverManagement , config):
        self.wd = wd
        self.config = config
    
    def web_setup(self):
        self.open_google()
        self.open_yt()
        self.open_sauce()
    
    def open_sauce(self):
        self.wd.create_page(self.config['WEB_URL'],"SAUCE")
        
    def open_yt(self):
        self.wd.create_page(self.config['YT'], "YT")
    
    def open_google(self):
        self.wd.create_page(self.config['GOOGLE'], "GOOGLE")

File: SAUCEDEMO-Playwright/TypeScript/utils/AppFactory.ts
import { WebDriverManagement } from "./driverFactory";

/** AppFactory 
 * 
 * A centralized place for setup webs testing and register it in pages to be select and use later.
*/
export class AppFactory {
  constructor(public wd: WebDriverManagement) {}

  async websSetup() {
    await this.openSauce();
    await this.openGoogle();
  }
  async openSauce() {
    await this.wd.createPage(this.wd.config.webURL, "SAUCE");
  }
  async openGoogle() {
    await this.wd.createPage(this.wd.config.google, "GOOGLE");
  }
  
}
